pages/dashboard/index.tsx

```diff
@@ -6,59 +6,33 @@ import NextEventCard from "../../components/common/NextEventCard/NextEventCard";
 import { parseCookies } from "nookies";
 import { GetServerSidePropsContext } from "next";
 import { Common } from "../../constants/common";
+import { fetchWrapper } from "../../util/httpWrapper";
 
-function DashboardHome({username, photo}: any) {
-    // dummy leaderboard
-    const leaderboard = [
-        {
-            name: "Harshpreet Singh Johar",
-            designation: "Chairperson",
-            dp: "https://avatars.githubusercontent.com/u/50266759?v=4",
-            xp: 70,
-        },
-        {
-            name: "Ishwarendra Jha",
-            designation: "Treasurer",
-            dp: "",
-            xp: 49,
-        },
-        {
-            name: "Akash Rout",
-            designation: "EB Memeber",
-            dp: "",
-            xp: 20,
-        },
-        {
-            name: "Priyanka Soni",
-            designation: "EB Member",
-            dp: "",
-            xp: 18,
-        },
-        {
-            name: "Pranav Sharma",
-            designation: "IB Member",
-            dp: "",
-            xp: 13,
-        },
-    ];
+function DashboardHome({ username, photo, leaderboard}: any) {    
 
     return (
         <DashboardLayout
             title="Dashboard | ACM at PEC"
             heading={<div className={styles.title}>Hey {username},</div>}
         >
             <div className={styles.trending}>
-                
+
                 <h2 className={styles.header}>What&#39;s going on?</h2>
                 <Trending trendingType="home" />
             </div>
             <div className={styles.details}>
                 <div className={styles.contributors}>
                     <h2 className={styles.header}>Top contributors</h2>
                     <div className={styles.leaderboard}>
-                        {leaderboard.map((user, index) => {
-                            return <LeaderboardItem user={user} key={index} />;
-                        })}
+                        {
+                            leaderboard &&
+                            leaderboard
+                                .filter((user: any) => user.xp > 0)
+                                .sort((user1: any, user2: any) => user2.xp - user1.xp)
```

Comment:
Just show first 5 here, not the full list

Reply:
Sure will do it

Reply:
Api returns first 5 only right now btw, hv made issue to make api pageable

Reply:
main lb page also gets only 5? @ken1000minus7

Reply:
yeah same endpoint hai, endpoint mei changes karne ke baad dashboard mei 5 ki limit kardenge

Reply:
alright, makes sense